gcpkms/keysource.go

```diff
@@ -243,37 +246,62 @@ func (key *MasterKey) newKMSClient() (*kms.KeyManagementClient, error) {
 	case key.credentialJSON != nil:
 		opts = append(opts, option.WithCredentialsJSON(key.credentialJSON))
 	default:
-		credentials, err := getGoogleCredentials()
-		if err != nil {
-			return nil, err
-		}
+		credentials, errCredentialsFile := getGoogleCredentials()
 		if credentials != nil {
 			opts = append(opts, option.WithCredentialsJSON(credentials))
+			break
+		}
+
+		atCredentials, errCredentialsToken := getGoogleOAuthTokenFromEnv()
+		if atCredentials != nil {
+			opts = append(opts, option.WithTokenSource(atCredentials))
+		}
+
+		if errCredentialsFile != nil && errCredentialsToken != nil {
+			return nil, fmt.Errorf("credentials: failed to get credentials for gcp kms, add default credentials or oauth access token from env")
```

Comment:
```suggestion
			return nil, fmt.Errorf("credentials: none found: missing %q and/or %q environment variables", SopsGoogleCredentialsEnv, SopsGoogleCredentialsOAuthTokenEnv)
```

makes it actionable.

Reply:
Also: checking the errors is theoretically less fool proof than checking if `credentials` and `atCredentials` are both `nil`, because that's the condition to actually have them configured.

Reply:
With the other changes suggested, I'd remove the `if` condition here completely.

Reply:
PTAL at the set of my suggestions that are still open.